commands/src/main/java/discord4j/commands/BaseCommand.java

```diff
@@ -0,0 +1,21 @@
+package discord4j.commands;
+
+import discord4j.core.event.domain.message.MessageCreateEvent;
+import reactor.core.publisher.Mono;
+
+/**
+ * Very simple command abstraction.
+ */
+@FunctionalInterface
+public interface BaseCommand {
```

Comment:
There are likely to be command wrappers over this would like the Command namespace. While clashes are not a technical problem, it may lead to end user confusion.

Reply:
In convention, typically when naming interfaces you refer to either a singular definition if you're not extending or add additional adjectives or nouns when extending or implementing. For example, `List` refers to a representation of a `List` ADT, while additional adjectives or nouns such as `Array` or `Abstract` is prefixed to denote the specific implementation or more quantify the abstraction, respectively.

`Base` in every sense that I have seen it refers as to a "baseline" implementation. It's a minimal implementation, a synonym I see that is more common but I feel represents the same thing is `Basic`, which this does not represent. If implementations are using `Command` namespace they are not being specific enough with their naming and I feel that causes *more* confusion than anything. Why should `Command` an implementation of `BaseCommand`, in every sense and reasonable way that I can conceive, it should be the other way arround. `Command` being the baseline interface, while a `BaseCommand` can be, I don't know, a basic command that has some default actions that other implementations should extend, or whatever. Point is, I don't believe this naming is justifiable.